src/PsrLogAdapter.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/*
+ * Copyright 2023-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB;
+
+use MongoDB\Driver\Monitoring\LogSubscriber;
+use MongoDB\Exception\UnexpectedValueException;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use SplObjectStorage;
+
+use function MongoDB\Driver\Monitoring\addSubscriber;
+use function MongoDB\Driver\Monitoring\removeSubscriber;
+use function sprintf;
+
+final class PsrLogAdapter implements LogSubscriber
```

Comment:
Ultimately, I think this entire class can be `@internal`, but I'm not going to do so now. The only user-facing API here is add/removeLogger, and I think those might be better presented as [functions](https://www.mongodb.com/docs/php-library/current/reference/functions/) like `with_transaction()`. Doing so would make it easier to keep this class out of view.

Also, the [Logging spec](https://www.mongodb.com/docs/php-library/current/reference/functions/) requires an API for configuring logging at a component level, which we presently can't implement because libmongoc itself doesn't. In the meantime I've asked the C team about the timeline for [CDRIVER-3775](https://jira.mongodb.org/browse/CDRIVER-3775) and likelihood that it would actually comply with the spec (since `mongoc-log.c` has several design conflicts).

Let's focus on getting https://github.com/mongodb/mongo-php-driver/pull/1395 merged first and I'll see what assurances I can extract from the C team to help inform the design we'll need in PHPLIB to comply with the spec.

Also, while we _could_ implement command and SDAM logging on our own using PHPC subscribers, that would set us up for duplicate logging once libmongoc gets around to implementing the spec and starts emitting its own messages. And since we can't prevent folks from using newer PHPC versions with older PHPLIB versions, there wouldn't be an easy way to work around the duplication.